### PR TITLE
Move Jackson Guice adapters into io.druid

### PR DESCRIPTION
**Pull request by drcrallen**
* Removes access to protected methods in com.fasterxml
* Eliminates druid-common's use of foreign package com.fasterxml

In the case where druid is loaded as a library in an framework that already loads Jackson, if the fasterxml classes and the druid classes are loaded by different classloaders, the protected method access will fail on the druid side. This PR prevents such protected method access by using public methods which were probably not present when the files were first written.

A MORE COMPLETE solution would probably be to migrate to https://github.com/FasterXML/jackson-module-guice at some point in the near future.

### CODE REVIEW

File: common/src/main/java/io/druid/guice/GuiceAnnotationIntrospector.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+{
+  @Override
+  public Object findInjectableValueId(AnnotatedMember m)
+  {
+    if (m.getAnnotation(JacksonInject.class) == null) {
+      return null;
+    }
+
+    Annotation guiceAnnotation = null;
+    for (Annotation annotation : m.annotations()) {
```
Comment (drcrallen) [inline common/src/main/java/io/druid/guice/GuiceAnnotationIntrospector.java]: This line did change, no other changes to the actual logic.

**Comment by gianm** [comment]
`DruidCoordinatorTest.testCoordinatorRun:349 expected:<1> but was:<0>`

**Comment by gianm** [comment]
:+1: 

Looks like this was added in 2.3 by: https://github.com/FasterXML/jackson-databind/commit/5309174c08ada60b2aaf89cccf264e76348ec1d1

**Comment by himanshug** [comment]
:+1: (after travis)

**Pull request merged**